pallets/parachain-staking/src/lib.rs

```diff
@@ -1702,6 +1867,79 @@ pub mod pallet {
 				rewardable_delegations,
 			}
 		}
+
+		/// This function exists as a helper to delegator_bond_more & auto_compound functionality.
+		/// Any changes to this function must align with both user-initiated bond increases and
+		/// auto-compounding bond increases.
+		/// Any feature-specific preconditions should be validated before this function is invoked.
+		/// Any feature-specific events must be emitted after this function is invoked.
+		pub fn delegation_bond_more_without_event(
+			delegator: T::AccountId,
+			candidate: T::AccountId,
+			more: BalanceOf<T>,
+		) -> Result<bool, sp_runtime::DispatchError> {
+			ensure!(
+				!Self::delegation_request_revoke_exists(&candidate, &delegator),
+				Error::<T>::PendingDelegationRevoke
+			);
+			let mut state = <DelegatorState<T>>::get(&delegator).ok_or(Error::<T>::DelegatorDNE)?;
+			state.increase_delegation::<T>(candidate.clone(), more)
+		}
+
+		/// Mint a specified reward amount to the beneficiary account. Emits the [Rewarded] event.
+		fn mint(amt: BalanceOf<T>, to: T::AccountId) {
+			if let Ok(amount_transferred) = T::Currency::deposit_into_existing(&to, amt) {
+				Self::deposit_event(Event::Rewarded {
+					account: to.clone(),
+					rewards: amount_transferred.peek(),
+				});
+			}
+		}
+
+		/// Mint and compound delegation rewards. The function mints the amount towards the
+		/// delegator and tries to compound a specified percent of it back towards the delegation.
+		/// If a scheduled delegation revoke exists, then the amount is only minted, and nothing is
+		/// compounded. Emits the [Compounded] event.
+		fn mint_and_compound(
+			amt: BalanceOf<T>,
+			compound_percent: Percent,
+			candidate: T::AccountId,
+			delegator: T::AccountId,
+		) {
+			if let Ok(amount_transferred) =
+				T::Currency::deposit_into_existing(&delegator, amt.clone())
+			{
```

Comment:
The error handling here seems a bit vague. Maybe the entire `fn mint_and_compound` should return a `Result`?

But the errors seem impossible...

`deposit_into_existing()` can only error if: `If who doesn’t exist, nothing is done and an Err returned.`

And `delegation_bond_more_without_event()` looks like it can only error if the delagator didn't exist, which should be impossible.

Reply:
I think this also stems from the fact, that we cannot do anything with the error here, sure I can bubble it up and log it in `pay_one_collator_reward` but it would still be a log line.

> `deposit_into_existing()` can only error if: If `who` doesn’t exist.
This can very well happen if a delegator account is somehow deleted (they unbond, withdraw everything, substrate cleans it up).

> `delegation_bond_more_without_event()` looks like it can only error if the delagator didn't exist
Which can also happen, if the delegator has revoked their delegation by the time we paid out rewards.

Reply:
Thanks for thinking about it. I agree with all of your points.